noir_stdlib/src/sha512.nr

```diff
@@ -0,0 +1,192 @@
+    
+    let mut out_h: [u64; 8] = hash;
+    let w = sha_w(msg);
+    for j in 0..80
+    {
+        let t1 = out_h[7] + sha_bigma1(out_h[4]) + sha_ch(out_h[4], out_h[5], out_h[6])
+            + K[j] + w[j];
+        let t2 = sha_bigma0(out_h[0]) + sha_maj(out_h[0], out_h[1], out_h[2]);
+        out_h[7] = out_h[6];
+        out_h[6] = out_h[5];
+        out_h[5] = out_h[4];
+        out_h[4] = out_h[3] + t1;
+        out_h[3] = out_h[2];
+        out_h[2] = out_h[1];
+        out_h[1] = out_h[0];
+        out_h[0] = t1 + t2;
+    };
+    
+    out_h
+}
+
+// Convert 128-byte array to array of 16 u64s
+fn msg_u8_to_u64(msg: [u8; 128]) -> [u64; 16]
+{
+    let mut msg64: [u64; 16] = [0; 16];
+
+    for i in 0..16
+    {
+        for j in 0..8
+        {
+            msg64[15 - i] = (msg64[15 - i] << 8) + msg[128 - 8*(i + 1) + j] as u64;
+        };
+    };
+    
+    msg64
+}
+
+// SHA-512 hash function
+fn digest(msg: [u8]) -> [u8; 64]
+{
+    let mut msg_block: [u8; 128] = [0; 128];
+    let mut h: [u64; 8] = [7640891576956012808, 13503953896175478587, 4354685564936845355, 11912009170470909681, 5840696475078001361, 11170449401992604703, 2270897969802886507, 6620516959819538809]; // Intermediate hash, starting with the canonical initial value
+    let mut c: [u64; 8] = [0; 8]; // Compression of current message block as sequence of u64
+    let mut out_h: [u8; 64] = [0; 64]; // Digest as sequence of bytes
+    let mut i = 0; // Message byte pointer
+
+    for m in msg
```

Comment:
Unfortunately this does not work inside the std::lib, could you rewrite it using:
`for k in 0..8`